R-packages/covidcast/R/plot.R

```diff
@@ -254,90 +253,107 @@ plot_choro = function(x, time_value = NULL, include = c(), range,
   geo = df$geo
   names(val) = geo
   
-  # Make background layer for MSA and HRR maps which are incomplete
-  if ((attributes(x)$metadata$geo_type == "msa") ||
-      (attributes(x)$metadata$geo_type == "hrr")) {
+  # Make background layer for maps.  For states view this isn't
+  # necessary but it just hides in the background
+  map_df = sf::st_read(system.file(
+    "shapefiles/state/cb_2019_us_state_5m.shp",
+    package = "covidcast"))
+  background_crs = sf::st_crs(map_df)
+  map_df$STATEFP <- as.character(map_df$STATEFP)
+  map_df = map_df %>% dplyr::mutate(
+    is_alaska = STATEFP == '02',
+    is_hawaii = STATEFP == '15',
+    is_pr = STATEFP == '72',
+    is_state = as.numeric(STATEFP) < 57)
+
+  # Set megacounty colors here
+  if(attributes(x)$metadata$geo_type == "county"){
+    map_df = map_df %>% dplyr::mutate(
+      color = ifelse(paste0(STATEFP,"000") %in% geo,
+              col_fun(val[paste0(STATEFP, "000")], alpha = alpha),
+              missing_col))
+  }
+  # Else, just set background to missing color
+  else {
+     map_df = map_df %>% dplyr::mutate(color = missing_col)
+  }
+
+  if (length(include) > 0) {
+    map_df = map_df %>% dplyr::filter(.$STUSPS %in% include)
+  }
+
+  main_df = shift_main(map_df)
+  hawaii_df = shift_hawaii(map_df)
+  alaska_df = shift_alaska(map_df)
+  pr_df = shift_pr(map_df)
+
+  main_col = main_df$color
+  hawaii_col = hawaii_df$color
+  alaska_col = alaska_df$color
+  pr_col = pr_df$color  
+
+  aes = ggplot2::aes
+  geom_args = list()
+  geom_args$color = border_col
+  geom_args$size = border_size
+  geom_args$mapping = aes(geometry=geometry)
+
+  geom_args$fill = main_col
+  geom_args$data = main_df
+  back_main_layer = do.call(ggplot2::geom_sf, geom_args)
+  geom_args$fill = pr_col
+  geom_args$data = pr_df
+  back_pr_layer = do.call(ggplot2::geom_sf, geom_args)
+  geom_args$fill = hawaii_col
+  geom_args$data = hawaii_df
+  back_hawaii_layer = do.call(ggplot2::geom_sf, geom_args)
+  geom_args$fill = alaska_col
+  geom_args$data = alaska_df
+  back_alaska_layer = do.call(ggplot2::geom_sf, geom_args)
+
+  # Create the choropleth colors for counties
+  if (attributes(x)$metadata$geo_type == "county") {
     map_df = sf::st_read(system.file(
-      "shapefiles/state/cb_2019_us_state_5m.shp",
+      "shapefiles/county/cb_2019_us_county_5m.shp",
       package = "covidcast"))
-    background_crs = sf::st_crs(map_df)
     map_df$STATEFP <- as.character(map_df$STATEFP)
-    map_df = map_df %>% dplyr::mutate(
+    map_df$GEOID <- as.character(map_df$GEOID)
+    # Get rid of unobserved counties and megacounties
+    # Those are taken care of by background layer
+    # Then set color for those observed counties
+    map_df = map_df %>% dplyr::filter(
+      (GEOID %in% geo) & !(COUNTYFP == "000")
```

Comment:
This formatting is weird since the call to `filter()` should fit on one line.

Reply:
Better?